packages/dev/test/provider/RelayProvider.test.ts

```diff
@@ -517,4 +526,107 @@ contract('RelayProvider', function (accounts) {
+        const signature = await web3eph.eth.sign(message, account)
+        const recover = getEcRecoverMeta(message, signature)
+        const onChainRecover = await testUtils._ecrecover(message, signature)
+        assert.equal(onChainRecover.toLowerCase(), account.toLowerCase(), 'on-chain ecrecover failed')
+        assert.equal(recover.toLowerCase(), account.toLowerCase(), 'off-chain ecrecover failed')
+      })
+
+      describe('eth_signTransaction', function () {
+        it('should sign using ephemeral key', async function () {
+          const transactionConfig: TransactionConfig = {
+            from: account,
+            to: account,
+            nonce: 0,
+            gasPrice: 1e9,
+            gas: 1000000,
+            value: 0,
+            data: '0xdeadbeef'
+          }
+          const transactionConfig1559: TransactionConfig = {
+            from: account,
+            to: account,
+            nonce: 1,
+            maxFeePerGas: 1e9,
+            gas: 1000000,
+            maxPriorityFeePerGas: 1,
+            value: 0,
+            data: '0xdeadbeef'
+          }
+          const res = await web3eph.eth.signTransaction(transactionConfig)
+          const res1559 = await web3eph.eth.signTransaction(transactionConfig1559)
+          await web3.eth.sendTransaction({ from: accounts[0], to: account, value: 1e18 })
+          const rec = await web3.eth.sendSignedTransaction(res.raw)
+          const rec1559 = await web3.eth.sendSignedTransaction(res1559.raw)
+          assert.equal(rec.from.toLowerCase(), account.toLowerCase())
+          assert.equal(rec1559.from.toLowerCase(), account.toLowerCase())
+          assert.equal(rec.gasUsed, 21064)
+          assert.equal(rec.gasUsed, 21064)
+        })
+
+        describe('eth_signTypedData', function () {
+          // TODO: once ERC-712 is added to Web3.js update this test
+          it('should sign using ephemeral key', async function () {
+            interface Test712 extends MessageTypes {
+              TestValueType: MessageTypeProperty[]
+            }
+
+            const dataToSign: TypedMessage<Test712> = {
+              types: { EIP712Domain: EIP712DomainType, TestValueType: [{ name: 'testValue', type: 'string' }] },
+              primaryType: 'TestValueType',
+              domain: {
+                name: 'domainName',
+                version: 'domainVer',
+                chainId: 1337,
+                verifyingContract: account
+              },
+              message: {
+                testValue: 'hello 712'
+              }
+            }
+            const paramBlock = {
+              method: 'eth_signTypedData',
+              params: [account, dataToSign],
+              jsonrpc: '2.0',
+              id: Date.now()
+            }
+            const promisified = new Promise<any>((resolve, reject) => {
+              (web3eph.currentProvider as HttpProvider).send(paramBlock, (error?: Error | null, result?: JsonRpcResponse): void => {
```

Comment:
its time we stop using this ugly and deprecated callback-mode, and start use "request"
(created an issue for that - https://opengsn.atlassian.net/browse/OG-816)